### DISCUSSION TITLE
Single Prisma Schema with multiple projects

### DISCUSSION THREAD

**Discussion by m-shariq** [prisma/prisma#19754] [Q&A]
**Issue**

Hey, I am using Prisma and working on multiple projects and using the same schema in all the projects. I want to update the schema from `one` project and the schema with types must be updated in `multiple projects.`

I have found several solutions on the internet such as `creating your package`. If I create my package, how I can manage schema separately for development, staging and production because I have to define the DB URL in `prisma.schema` file?

Is there any better approach for manage the schema from one repo and use everywhere `without copy-pasting the schema` in all other projects?

Thanks

**Comment by nurul3101** [comment]
Hello @m-shariq 👋 

> I have found several solutions on the internet such as creating your package

Yes, this would be the most straightforward approach for your use case. You can export Prisma Client and generated types in a custom npm package. That way, you can manage your prisma schema in a single place, but still use Prisma Client and generated types in multiple applications.

> If I create my package, how I can manage schema separately for development, staging and production because I have to define the DB URL in prisma.schema file

When you import the exported Prisma Client from your custom npm package, you can programmatically override the data source URL as mentioned in this guide: [Programmatically override a data source URL](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#programmatically-override-a-datasource-url)

Assuming you are using a PostgreSQL database, you can override the datasource URL like this:
```typescript
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient({
  datasources: {
    db: {
      url: 'postgresql://postgres:postgres@localhost:5432/staging?schema=public',
    },
  },
})
```

You can even use environment variables to pass the datasource URL, here's an example:

```ts
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient({
  datasources: {
    db: {
      url: process.env.STAGING_DATABASE_URL,
    },
  },
})
```

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by m-shariq** [reply]
Thank you @nurul3101. I've seen this and my question is, should I just add a random URL in my npm package and then override with the actual one?

**Reply by nurul3101** [reply]
Yes, as long as you are overriding the datasource URL, you can add a random URL, it won't matter.

**Reply by nurul3101** [reply]
Do let us know in case you still have any further queries. If not, please consider marking this discussion as Answered to indicate that this has been resolved.

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
